tools/setup_services.py

```diff
@@ -0,0 +1,60 @@
+import re
+import os
+from os.path import join as pjoin
+from collections import Counter
+
+from baselayer.app.env import load_env
+from baselayer.log import make_log
+
+
+log = make_log('baselayer')
+
+
+def copy_supervisor_configs():
+    env, cfg = load_env()
+
+    services = {}
+    for path in cfg['services.paths']:
+        if os.path.exists(path):
+            path_services = [d for d in os.listdir(path)
+                             if os.path.isdir(pjoin(path, d))]
+            services.update({s: pjoin(path, s) for s in path_services})
+
+    duplicates = [k for k, v in Counter(services.keys()).items() if v > 1]
+    if duplicates:
+        raise RuntimeError(f'Duplicate service definitions found for {duplicates}')
+
+    log(f'Discovered {len(services)} services')
+
+    disabled = cfg['services.disabled'] or []
+    enabled = cfg['services.enabled'] or []
+
+    both = set().union(disabled).intersection(enabled)
+    if both:
+        raise RuntimeError(
+            f'Invalid service specification: {both} in both enabled and disabled'
+        )
+
+    if disabled == '*':
+        disabled = services.keys()
+    if enabled == '*':
```

Comment:
Why does the enabled list get set as empty if all services are set to be enabled?

Reply:
The logic is a few lines down. It wouldn't hurt to set it to the full list, it's just that we start with the full list (so, that's the default), take away as necessary from disabled, then add back any specified services.

Since the default is all services, setting enabled to star is equivalent to not modifying the default.